compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1629,7 +1629,7 @@ impl<'a> Parser<'a> {
 
     /// Parses a declarative macro 2.0 definition.
     /// The `macro` keyword has already been parsed.
-    /// ```
+    /// ```ignore (not-rust)
```

Comment:
Maybe this should be `plain`? That way it won't get rust syntax highlighting either.

Reply:
Yeah, just `text` here is better than ignore.

Reply:
Alright, so all the grammar ones should be `text`? I thought the syntax highlighting looked ok on them personally but I can see why you wouldn't want it.

Reply:
I don't think we should mark things that aren't rust as rust, no. Anything using EBNF should be `text` for now (rustdoc currently only supports syntax highlighting for rust: https://github.com/rust-lang/rust/issues/56213).

Reply:
Oh ok so we just shouldn't mark things that aren't rust as `ignore` just for the (correct-ish looking but technically incorrect) highlighting?
The main classes of things I did that for are:

- ebnf (which I'll go make ` ```ebnf ` now)
- MIR, which I marked as `ignore (MIR)` (forgot I separated those from the other `(not-rust)` ones)
- chalk style type stuff, e.g. 
```
forall (U: Trait + ?Sized) {
    if (Self: Unsize<U>) {
        Receiver: DispatchFromDyn<Receiver[Self => U]>
    }
}
```
and
```
for<...> <T as Trait>::U == V
```
- plus a few odd ones out (shall we make them all text? am I misunderstanding some of them?)
  - https://github.com/rust-lang/rust/pull/96094/files/cd7734e958a70641c9492aa2e33572a4de4a83be#diff-5e6fdd07e3ea2b5e20c9fdd3914d1e781ecb4ac4f5349cb2ce33732da8d8a597R173-R175
  - https://github.com/rust-lang/rust/pull/96094/files/cd7734e958a70641c9492aa2e33572a4de4a83be#diff-ee45792072b570212b473de3da83a965c3ce9b74a86109d27e818f7486d8180aR13-R16
  - https://github.com/rust-lang/rust/pull/96094/files/cd7734e958a70641c9492aa2e33572a4de4a83be#diff-a75cbce2712ef8e4b4013fc2189dcd0890974a11880e02ca16a493228add9619R1896
  - https://github.com/rust-lang/rust/pull/96094/files/cd7734e958a70641c9492aa2e33572a4de4a83be#diff-1677e6785e45881b13f3477c96f0e619c7ee07cd0fc54fb91b877cf82b3df22fR9

Should I make the MIR and chalk ones ` ```mir ` and ` ```chalk ` or ` ```text` or are they ok as-is because they're rust-y enough?

Reply:
Nah, I'd say just leave them as is. The main thing is to get the tests passing.